app/file.go

```diff
@@ -593,6 +593,13 @@ func (a *App) UploadFileX(channelId, name string, input io.Reader,
 		return t.fileinfo, aerr
 	}
 
+	// attempt to extract width & height out of svg
```

Comment:
This should be moved to preprocessImage. There is a similar exception made there for GIFs.

Reply:
Hey @levb, I didn't end up putting this in `preprocessImage()` because SVG is not actually a raster-based image format and is kicked out at the beginning of that function by an `image.DecodeConfig` error. The rest of that function processes raster images (like creating previews for larger files etc.), which wouldn't really work for SVG's. 

While the mime type does class it as an image (`image/svg+xml`) and it obviously displays like an image in an HTML `<img />` tag, it is actually vector data represented using XML so can't be decoded and manipulated like a .gif, .jpg etc.

What we need from SVG's to better treat them like images for the new scrolling behaviour currently being work on, is their dimensions. In order to get the dimensions, the xml in the file needs to be parsed to read several xml attributes (viewBox, width & height), which means the file needs to have already been loaded in so I either have to load it myself (duplicating effort), or piggy back off of the existing `readAll()` effort that runs just after `preprocessImage()`.

That said, I guess it would at least make sense to move it up before `runPlugins()` so that plugins could maybe affect the width/height of SVG's if they need?

Also, maybe we should prevent SVG's from running through `preprocessImage()` altogether as they will always be immediately ejected given the current implementation?

Thoughts? Let me know if I misunderstand! :)

Reply:
I think SVG are images, and we should treat them as such.

> it is actually vector data.
Understood. We can still disable the preview/thumbnail functionality, like we do for animated GIFs, https://github.com/mattermost/mattermost-server/blob/7150b9aada568f5fd9f674c27beaa5890bb53183/app/file.go#L712

> we need from SVG's ..., is their dimensions
Precisely. Which is what preprocessImages generally does - extracts metadata for images and populates FileInfo. That SVGs are not supported by an `image/xxx` package in Go shouldn't matter. 3/5 I recommend invoking the SVG parsing from there. Obviously, none of the existing code in that function applies, so it'd be an `if SVG { do something and return }`.

postprocessImages should skip them altogether.

Alternatively... why not implement this as a plugin that implements `FileWillBeUploaded` hook and updates the FileInfo? :)

Reply:
Pretty sure this is core functionality, not a plugin feature that will be enabled and disabled.

Reply:
@hmhealy, can you shed some light into the PostMetadata needs here? I’m only able to provide a review of the code changes, and don’t have context into the larger pipeline.